controllers/cards.js

```diff
@@ -8,41 +8,49 @@ const getCards = (req, res) => {
 };
 
 const createCard = (req, res) => {
-  const {
-    name, link, owner, likes, createAt,
-  } = req.body;
+  const { name, link } = req.body;
 
-  Card.create({
-    name, link, owner, likes, createAt,
-  })
+  Card.create({ name, link, owner: req.user })
     .then((card) => res.send({ data: card }))
     .catch(() => res.status(500).send({ message: 'Произошла ошибка' }));
 };
 
 const deleteCard = (req, res) => {
   Card.findById(req.params.id)
     .then((card) => {
-      if (req.user._id !== card.owner._id) {
-        return Promise.reject(new Error('Отсутствуют права на редактирование!'));
+      if (!card) {
+        return res.status(404).send({ message: 'Карточка с таким id не найдена' });
       }
-      return Card.findByIdAndRemove(req.params.id);
+
+      if (!card.owner.equals(req.user._id)) {
+        return res.status(403).send({ message: 'Отсутствуют права на редактирование!' });
+      }
+
+      return Card.remove(req.params.id)
```

Comment:
Нужно исправить: некорректно используется вызов метода `remove()`.
Тут ошибка в коде:
```bash
{ ObjectParameterError: Parameter "filter" to remove() must be an object, got 5e5e95a3f3bb02ffe808305d
    at new ObjectParameterError (/Users/jmi/work/ya-reviews/serv/node_modules/mongoose/lib/error/objectParameter.js:25:11)
    at model.Query.Query.remove (/Users/jmi/work/ya-reviews/serv/node_modules/mongoose/lib/query.js:2629:16)
    at Function.remove (/Users/jmi/work/ya-reviews/serv/node_modules/mongoose/lib/model.js:1908:13)
    at Card.findById.then (/Users/jmi/work/ya-reviews/serv/controllers/cards.js:29:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  message:
   'Parameter "filter" to remove() must be an object, got 5e5e95a3f3bb02ffe808305d',
  name: 'ObjectParameterError' }
```
Метод `.findById()` возвращает `card` в таком формате, который требует метод `.remove()`, поэтому просто передаем его:
```suggestion
      return Card.remove(card)
```